core/src/main/java/edu/wpi/grip/core/operations/composite/PublishVideoOperation.java

```diff
@@ -0,0 +1,189 @@
+    private static final byte[] MAGIC_NUMBER = {0x01, 0x00, 0x00, 0x00};
+
+    private final Object imageLock = new Object();
+    private RenderedImage image = null;
+    private Optional<Thread> serverThread = Optional.empty();
+    private volatile float compressionQuality = 0.5f;
+    private volatile boolean connected = false;
+    private volatile int numSteps = 0;
+
+    /**
+     * Listens for incoming connections on port 1180 and writes JPEG data whenever there's a new frame.
+     */
+    private final Runnable runServer = () -> {
+        // Loop forever (or at least until the thread is interrupted).  This lets us recover from the dashboard
+        // disconnecting or the network connection going away temporarily.
+        while (!Thread.currentThread().isInterrupted()) {
+            ServerSocket serverSocket = null;
+            Socket socket = null;
+
+            try {
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+                ImageWriter jpegWriter = ImageIO.getImageWritersByFormatName("jpeg").next();
+                jpegWriter.setOutput(ImageIO.createImageOutputStream(baos));
+
+                ImageWriteParam jpegWriteParam = jpegWriter.getDefaultWriteParam();
+                jpegWriteParam.setCompressionMode(ImageWriteParam.MODE_EXPLICIT);
+
+                serverSocket = new ServerSocket(PORT);
+                logger.info("Starting camera server");
+
+                while (!serverSocket.isClosed() && !Thread.currentThread().isInterrupted()) {
+                    socket = serverSocket.accept();
+                    logger.info("Got connection from " + socket.getInetAddress());
+                    connected = true;
+
+                    DataOutputStream socketStream = new DataOutputStream(socket.getOutputStream());
+
+                    while (!socket.isClosed() && !Thread.currentThread().isInterrupted()) {
+                        baos.reset();
+
+                        // Wait for the main thread to put a new image. This happens whenever perform() is called with
+                        // a new input.
+                        synchronized (imageLock) {
+                            imageLock.wait();
+                            jpegWriteParam.setCompressionQuality(compressionQuality / 100.0f);
+                            jpegWriter.write(null, new IIOImage(image, null, null), jpegWriteParam);
+                        }
+
+                        // The FRC dashboard image protocol consists of a magic number, the size of the image data,
+                        // and the image data itself.
+                        socketStream.write(MAGIC_NUMBER);
+                        socketStream.writeInt(baos.size());
+                        baos.writeTo(socketStream);
+                    }
+                }
+            } catch (IOException e) {
+                logger.log(Level.WARNING, e.getMessage(), e);
+            } catch (InterruptedException e) {
```

Comment:
You must set the thread's interrupted flag here. The default thread handler depends on it being set.